include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -450,14 +448,14 @@ TridiagResult<T, Device::CPU> BandToTridiag<Backend::MC, D, T>::call_L(
   for (SizeType i = 0; i < max_workers; ++i)
     workers.emplace_back(SweepWorker<T>(size, b));
 
-  auto init_sweep = [a_ws](SizeType sweep, PromiseGuard<SweepWorker<T>> worker) {
-    worker.ref().startSweep(sweep, *a_ws);
+  auto init_sweep = [a_ws](SizeType sweep, SweepWorker<T>& worker) {
+    worker.startSweep(sweep, *a_ws);
   };
-  auto cont_sweep = [a_ws, b](SizeType nr_steps, PromiseGuard<SweepWorker<T>> worker,
+  auto cont_sweep = [a_ws, b](SizeType nr_steps, SweepWorker<T>& worker,
                               matrix::Tile<T, Device::CPU>&& tile_v, TileElementIndex index) {
     for (SizeType j = 0; j < nr_steps; ++j) {
-      worker.ref().compactCopyToTile(tile_v, index + TileElementSize(j * b, 0));
-      worker.ref().doStep(*a_ws);
+      worker.compactCopyToTile(tile_v, index + TileElementSize(j * b, 0));
+      worker.doStep(*a_ws);
```

Comment:
TODO, check with @albestro if this is really ok.

Reply:
There's a slight semantic difference between the old and the new behaviour and whether or not it's a problem depends on what comes after `cont_sweep`. Though I see @rasolca has made the same change (to unwrap _all_ `PromiseGuards`) in https://github.com/eth-cscs/DLA-Future/pull/698/files, so that's reassuring.

The difference is that if `PromiseGuard<T>` is taken by value by the function it will be released at the end of the function scope. If the `PromiseGuard<T>` is unwrapped before being passed to the function it will live as long as the rvalue reference of it lives inside `then` (for CPU), which may be until the next `make_future`/`ensure_started`/`start_detached` (or other adaptor that releases previous operation states).